wire-ios-sync-engine/Source/UserSession/ZMUserSession/ZMUserSession.swift

```diff
@@ -524,11 +542,39 @@ public final class ZMUserSession: NSObject {
             }
         }
 
-        if let clientId = selfUserClient?.safeRemoteIdentifier.safeForLoggingDescription {
-            WireLogger.authentication.addTag(.selfClientId, value: clientId)
+        if let selfUserClient {
+            WireLogger.authentication.addTag(
+                .selfClientId,
+                value: selfUserClient.safeRemoteIdentifier.safeForLoggingDescription
+            )
+
+            // Create and perform sync if there is a self client.
+            if let selfClientID = selfUserClient.remoteIdentifier {
+                setUpSyncAgent(clientID: selfClientID)
+
+                Task {
+                    do {
+                        try await syncAgent?.performSyncIfNeeded()
+                    } catch {
+                        WireLogger.sync.error("failed to perform sync on session setup: \(String(describing: error))")
+                    }
+                }
+            }
         }
```

Comment:
@johnxnguyen I mean here is basically the same as:

```

        // The client was just registered and still needs to perform the
        // initial sync.
        if let selfClientID = userClient.remoteIdentifier {
            setUpSyncAgent(clientID: selfClientID)
            triggerInitialSync()
        }
```

Reply:
ok, done 1c353474f4f619110c6450508ac31182bda0167c